### ISSUE TITLE
[PyMcaIO] Compatibility with h5py 3.0

### ISSUE THREAD

**Issue by vasole** [vasole/pymca#757]
The tests do not pass with h5py 3.0 It seems all string datasets are read as bytes now

@woutdenolf Keep an eye on it for the bliss NexusWriter

**Comment by vasole** [comment]
```
======================================================================
ERROR: testHdf5Uri (ConfigDictTest.testConfigDict)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/ConfigDictTest.py", line 166, in testHdf5Uri
    readInstance.read(uri)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/PyMcaIO/ConfigDict.py", line 118, in read
    config = StringIO(h5[path][()])
TypeError: initial_value must be str or None, not bytes

======================================================================
FAIL: testNxExtStringAttribute (NexusUtilsTest.testNexusUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 186, in testNxExtStringAttribute
    self._checkStringTypes(attribute=True, raiseExtended=False)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 276, in _checkStringTypes
    self.assertEqual(type(value), type(expectedValue), msg=msg)
AssertionError: <class 'str'> != <class 'bytes'> : ext(scalar) <class 'str'> instead of <class 'bytes'>

======================================================================
FAIL: testNxExtStringDataset (NexusUtilsTest.testNexusUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 191, in testNxExtStringDataset
    self._checkStringTypes(attribute=False, raiseExtended=False)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 276, in _checkStringTypes
    self.assertEqual(type(value), type(expectedValue), msg=msg)
AssertionError: <class 'bytes'> != <class 'str'> : ascii(scalar) <class 'bytes'> instead of <class 'str'>

======================================================================
FAIL: testNxStringDataset (NexusUtilsTest.testNexusUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 181, in testNxStringDataset
    self._checkStringTypes(attribute=False, raiseExtended=True)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/PyMca5/tests/NexusUtilsTest.py", line 276, in _checkStringTypes
    self.assertEqual(type(value), type(expectedValue), msg=msg)
AssertionError: <class 'bytes'> != <class 'str'> : ascii(scalar) <class 'bytes'> instead of <class 'str'>

----------------------------------------------------------------------
Ran 91 tests in 303.957s

FAILED (failures=3, errors=1, skipped=3)
```

**Comment by woutdenolf** [comment]
I just checked and the writer tests pick up on this, so we will notice for sure. The silx CI is also broken because of this.

**Comment by woutdenolf** [comment]
For reference
silx: https://github.com/silx-kit/silx/issues/3218
bliss: https://gitlab.esrf.fr/bliss/bliss/-/issues/2187

**Comment by vasole** [comment]
closed by #758
